runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -54,6 +56,55 @@ void waitForWorkflow(UUID workflow, ManagementClient client, Duration timeout,
         throw new TimeoutException();
     }
 
+    /**
+     * Remove a node from the cluster.
+     *
+     * @param endpointToRemove Endpoint of the node to be removed from the cluster.
+     * @param retry the number of times to retry a workflow if it fails
+     * @param timeout total time to wait before the workflow times out
+     * @param pollPeriod the poll interval to check whether a workflow completed or not
+     * @throws WorkflowException throws a workflow exception on failure
+     */
+    public void removeNode(@Nonnull String endpointToRemove, int retry,
+                           @Nonnull Duration timeout, @Nonnull Duration pollPeriod) {
+        for (int x = 0; x < retry; x++) {
+            runtime.invalidateLayout();
```

Comment:
There seems to be a lot of common code between add and remove node. Can we unify it so that we only have one set of issues to deal with.